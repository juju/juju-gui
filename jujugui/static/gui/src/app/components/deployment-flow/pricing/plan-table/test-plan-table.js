/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const DeploymentPlanTable = require('./plan-table');

describe('DeploymentPlanTable', () => {
  let applications, charms, listPlansForCharm;

  const renderComponent = (options = {}) => enzyme.shallow(
    <DeploymentPlanTable
      addNotification={options.addNotification || sinon.stub()}
      applications={options.applications || applications}
      charms={options.charms || charms}
      listPlansForCharm={options.listPlansForCharm || listPlansForCharm} />
  );

  beforeEach(() => {
    applications = [{
      getAttrs: sinon.stub().returns({
        icon: 'apache2.svg',
        charm: 'cs:apache2',
        id: 'apache2-123',
        name: 'apache2'
      })
    }, {
      getAttrs: sinon.stub().returns({
        charm: 'cs:mysql'
      })
    }];
    charms = {
      getById: sinon.stub()
    };
    charms.getById.withArgs('cs:apache2').returns({
      get: sinon.stub().withArgs('id').returns('cs:apache2'),
      hasMetrics: sinon.stub().returns(true)
    });
    charms.getById.withArgs('cs:mysql').returns({
      hasMetrics: sinon.stub().returns(false)
    });
    listPlansForCharm = sinon.stub().callsArgWith(1, null, [{
      description: 'The standard plan description',
      metrics: {
        memory: 'something'
      },
      price: '$3.25 per GB of RAM per month'
    }]);
  });

  it('can render', function() {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('can handle errors when getting plans', function() {
    const addNotification = sinon.stub();
    listPlansForCharm.callsArgWith(1, 'Uh oh!', null);
    renderComponent({addNotification});
    assert.equal(addNotification.callCount, 1);
    assert.deepEqual(addNotification.args[0][0], {
      title: 'Fetching plans failed',
      message: 'Fetching plans failed: Uh oh!',
      level: 'error'
    });
  });
});
