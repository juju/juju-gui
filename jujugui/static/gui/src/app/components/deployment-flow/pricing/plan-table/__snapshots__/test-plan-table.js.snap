// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentPlanTable can render 1`] = `
<div
  className="deployment-plan-table v1"
>
  <BasicTable
    headerClasses={
      Array [
        "deployment-plan-table__header-row",
      ]
    }
    headerColumnClasses={
      Array [
        "deployment-plan-table__header-column",
      ]
    }
    headers={
      Array [
        Object {
          "content": "Applications",
        },
        Object {
          "content": "Plan",
        },
        Object {
          "content": "Metered",
        },
        Object {
          "classes": Array [
            "u-align--right",
          ],
          "content": "Price",
        },
      ]
    }
    rowClasses={
      Array [
        "deployment-plan-table__row",
      ]
    }
    rowColumnClasses={
      Array [
        "deployment-plan-table__column",
      ]
    }
    rows={
      Array [
        Object {
          "columns": Array [
            Object {
              "content": <div>
                <img
                  alt="apache2"
                  className="deployment-plan-table__charm-icon"
                  src="apache2.svg"
                />
                <span
                  className="deployment-plan-table__charm-name"
                >
                  apache2
                </span>
              </div>,
            },
            Object {
              "content": <div>
                <p
                  className="deployment-plan-table__plan-description"
                >
                  The standard plan description
                </p>
              </div>,
            },
            Object {
              "content": <span
                className="deployment-plan-table__metered"
              >
                memory
              </span>,
            },
            Object {
              "classes": Array [
                "u-align--right",
              ],
              "content": "$3.25 per GB of RAM per month",
            },
          ],
          "key": "apache2-123",
        },
      ]
    }
    tableClasses={
      Array [
        "no-margin-bottom",
      ]
    }
  />
</div>
`;
